test/PhpSessionPersistenceTest.php

```diff
@@ -159,4 +161,186 @@ public function testPersistSessionIfSessionHasContents()
+
+        $control = sprintf('public, max-age=%d', $maxAge);
+        $expires = $response->getHeaderLine('Expires');
+        $expires = strtotime($expires);
+
+        $this->assertTrue($expires >= $expiresMin);
+        $this->assertTrue($expires <= $expiresMax);
+        $this->assertSame($response->getHeaderLine('Cache-Control'), $control);
+    }
+
+    public function testPersistSessionReturnsExpectedResponseWithCacheHeadersIfCacheLimiterIsPrivate()
+    {
+        $expire = 222;
+        $maxAge = 60 * $expire;
+
+        $this->startSession(null, [
+            'cache_limiter' => 'private',
+            'cache_expire'  => $expire,
+        ]);
+
+        $persistence = new PhpSessionPersistence();
+
+        $session  = new Session(['foo' => 'bar']);
+        $response = $persistence->persistSession($session, new Response());
+
+        $expires = PhpSessionPersistence::CACHE_PAST_DATE;
+        $control = sprintf('private, max-age=%d', $maxAge);
+
+        $this->assertSame($response->getHeaderLine('Expires'), $expires);
+        $this->assertSame($response->getHeaderLine('Cache-Control'), $control);
+    }
+
+    public function testPersistSessionReturnsExpectedResponseWithCacheHeadersIfCacheLimiterIsPrivateNoExpire()
+    {
+        $expire = 333;
+        $maxAge = 60 * $expire;
+
+        $this->startSession(null, [
+            'cache_limiter' => 'private_no_expire',
+            'cache_expire'  => $expire,
+        ]);
+
+        $persistence = new PhpSessionPersistence();
+
+        $session  = new Session(['foo' => 'bar']);
+        $response = $persistence->persistSession($session, new Response());
+
+        $control = sprintf('private, max-age=%d', $maxAge);
+
+        $this->assertSame($response->getHeaderLine('Expires'), '');
+        $this->assertSame($response->getHeaderLine('Cache-Control'), $control);
+    }
+
+    public function testPersistSessionReturnsExpectedResponseWithoutAddedHeadersIfAlreadyHasAny()
+    {
+        $this->startSession(null, [
+            'cache_limiter' => 'nocache',
+        ]);
+
+        $persistence = new PhpSessionPersistence();
+
+        $response = new Response('php://memory', 200, [
+            'Last-Modified' => gmdate(PhpSessionPersistence::HTTP_DATE_FORMAT),
+        ]);
+
+        $session  = new Session(['foo' => 'bar']);
+        $response = $persistence->persistSession($session, $response);
+
+        $this->assertFalse($response->hasHeader('Pragma'));
+        $this->assertFalse($response->hasHeader('Expires'));
+        $this->assertFalse($response->hasHeader('Cache-Control'));
+    }
+
+    public function testPersistSessionReturnsExpectedResponseWithLastModifiedHeader()
+    {
+        $this->startSession(null, [
+            'cache_limiter' => 'public',
+        ]);
+
+        // temporarily set SCRIPT_FILENAME to current file for testing
+        $scriptFilename = $_SERVER['SCRIPT_FILENAME'] ?? null;
+        $_SERVER['SCRIPT_FILENAME'] = __FILE__;
```

Comment:
With the changes I suggested earlier around usage of superglobals, this test will need to change.

What I didn't realize is that you're accessing it during `persistSession()` as well. I'm wondering if the script filename should be memoized in the Session instance, and removed during persistence?

Reply:
yes, see my previous reply...I don't think we need to remove it though whit it being refreshed at every request...and now that I think about it...should we just memoize it once....the running script is not going to change even in asynchronous contexts?

Reply:
Sure.

Reply:
....mhmmm....but mocked requests could contain a different SCRIPT_FILENAME....